src/workerd/api/streams/common.h

```diff
@@ -302,6 +302,15 @@ class ReadableStreamController {
   virtual kj::Maybe<PipeController&> tryPipeLock(jsg::Ref<WritableStream> destination) = 0;
 
   virtual void visitForGc(jsg::GcVisitor& visitor) {};
+
+  virtual jsg::Promise<kj::Array<byte>> readAllBytes(jsg::Lock& js,
+                                                     uint64_t limit = kj::maxValue) = 0;
+  virtual jsg::Promise<kj::String> readAllText(jsg::Lock& js,
+                                               uint64_t limit = kj::maxValue) = 0;
+  // Fully consumes the ReadableStream. If the stream is already locked to a reader or
+  // errored, the returned JS promise will reject. If the stream is already closed, the
+  // returned JS promise will resolve with a zero-length result. Importantly, this will
+  // lock the stream and will fully consume it.
```

Comment:
A little blurb about what `limit` does would be good.

Originally, the KJ equivalents of these functions didn't accept a `limit`, but we realized this was dangerous, and added them later. In order not to break existing code, we defaulted the parameter to `kj::maxValue`, even though ideally every caller should be forced to supply an appropriate limit.

Since these are new APIs that aren't used anywhere yet, I think it'd be wise to forego the default value.